src/modules/Elsa.OpenTelemetry/Middleware/OpenTelemetryTracingActivityExecutionMiddleware.cs

```diff
@@ -72,6 +73,31 @@ public async ValueTask InvokeAsync(ActivityExecutionContext context)
 
         if (!string.IsNullOrWhiteSpace(context.WorkflowExecutionContext.CorrelationId))
             span.SetTag("workflow.correlation_id", context.WorkflowExecutionContext.CorrelationId);
+
+        span.SetTag("workflow.instance.id", context.WorkflowExecutionContext.Id);
+        span.SetTag("workflow.definition.id", context.WorkflowExecutionContext.Workflow.Identity.DefinitionId);
+        span.SetTag("workflow.definition.version", context.WorkflowExecutionContext.Workflow.Identity.Version);
+
+        SetExtensions(context, span);
+    }
+
+    private void SetExtensions(ActivityExecutionContext context, Activity span)
+    {
+        var extensions = context.GetExtensionsMetadata();
+        if (extensions is not null)
+        {
+            foreach (var key in extensions.Keys)
+            {
+                span.SetTag($"activity.extensions.{ToOtelFormat(key)}", extensions[key]);
+            }
+        }
+    }
+
+    public string ToOtelFormat(string input)
+    {
+        var humanized = input.Humanize();
+        var correlationId = humanized.Replace(" ", "_").ToLowerInvariant();
+        return correlationId;
     }
```

Comment:
I'm not sure, but maybe it's worthwhile optimizing this with a single string allocation instead of two, considering that this code executes for each activity execution.

```csharp
public string ToOtelFormat(string input)
{
    if (string.IsNullOrWhiteSpace(input))
        return string.Empty;

    var sb = new StringBuilder(input.Length);

    foreach (var c in input)
    {
        if (char.IsWhiteSpace(c))
            sb.Append('_');
        else
            sb.Append(char.ToLowerInvariant(c));
    }

    return sb.ToString();
}
```